### DISCUSSION TITLE
there seems to be a bug here

### DISCUSSION THREAD

**Discussion by GeekTR** [LSPosed/LSPosed#2310] [Q&A]
I tried hooking the methods of class android.content.pm.IPackageManager$Stub$Proxy, but nothing works? But hooking the methods of other classes works fine.

**Comment by gab8192** [comment]
I don't know if it is possible for your exact problem, but you could hook the PackageManagerService instead. Then check whether the caller (Binder.getCallingUid()) is the target application, and in case, you know... do your thing

**Reply by GeekTR** [reply]
I know this, but in the PackageManangerService it is not possible to correctly distinguish the real package name of the app with the declared sharedUserId, because they have the same 'callingUid'
